### ISSUE TITLE
AMP-form textarea validation visible class not triggered for [visible-when-invalid]

### ISSUE THREAD

**Issue by Jessman5** [ampproject/amphtml#22237]
## What's the issue?

I have issues with showing the [visible-when-invalid] notification for a required textarea.
AMP html is valid and AMP-form module is included.

## How do we reproduce the issue?

this is my code 
```html
<form id="comment-write" method="post" action-xhr="https://ampbyexample.com/components/amp-form/submit-form-input-text-xhr" target="_top" custom-validation-reporting="show-all-on-submit">
    <fieldset>
        <textarea id="show-all-on-submit-comment" aria-label="x" placeholder="x" autoexpand required pattern="[A-Za-z0-9]{1,20}"></textarea>
        <span visible-when-invalid="valueMissing" validation-for="show-all-on-submit-comment">foo</span>
        <span visible-when-invalid="typeMismatch" validation-for="show-all-on-submit-comment">bar</span>
    </fieldset>

    <fieldset>
        <button type="reset" on="tap:comment-write.clear">Cancel</button>
        <button type="submit">Send</button>
    </fieldset>
</form>
```

I know that validation for textareas is quite hacky atm.. Though, in my understanding, it is supposed to show foo when I submit the form with an empty textarea. But it doesn't, since foo is only shown when `[visible-when-invalid]` gets the class `visible`.
Am I doing sth wrong here?

I tried a workaround with css setting `[visible-when-invalid]` to `display: none;` and to `display: block;` again, when the textarea gets the class `user-invalid`. But that leads to more issues like getting the foo notice even when I clicked into the textarea and clicking somewhere else without having typed sth into it.

## What browsers are affected?

All major browsers.

## Which AMP version is affected?

AMP-Version is 1904301721170

**Comment by torch2424** [comment]
Triaging to @cvializ , feel free to re-assign 😄

**Comment by Jessman5** [comment]
any progress on this so far?

**Comment by cvializ** [comment]
Not yet, but you are welcome to propose and implement a fix for it if you have capacity before we do
